pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -0,0 +1,348 @@
+
+	// Create the infra manifests.
+	for _, m := range infraManifests {
+		m.SetNamespace(capiutils.Namespace)
+		if err := cl.Create(ctx, m); err != nil {
+			return fileList, fmt.Errorf("failed to create infrastructure manifest: %w", err)
+		}
+		logrus.Infof("Created manifest %+T, namespace=%s name=%s", m, m.GetNamespace(), m.GetName())
+	}
+
+	// Pass cluster kubeconfig and store it in; this is usually the role of a bootstrap provider.
+	{
+		key := client.ObjectKey{
+			Name:      clusterID.InfraID,
+			Namespace: capiutils.Namespace,
+		}
+		cluster := &clusterv1.Cluster{}
+		if err := cl.Get(ctx, key, cluster); err != nil {
+			return fileList, err
+		}
+		// Create the secret.
+		clusterKubeconfig := clusterKubeconfigAsset.Files()[0].Data
+		secret := utilkubeconfig.GenerateSecret(cluster, clusterKubeconfig)
+		if err := cl.Create(ctx, secret); err != nil {
+			return fileList, err
+		}
+	}
+
+	// Wait for successful provisioning by checking the InfrastructureReady
+	// status on the cluster object.
+	var cluster *clusterv1.Cluster
+	{
+		if err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
+			Duration: time.Second * 10,
+			Factor:   float64(1.5),
+			Steps:    32,
+		}, func(ctx context.Context) (bool, error) {
+			c := &clusterv1.Cluster{}
+			if err := cl.Get(ctx, client.ObjectKey{
+				Name:      clusterID.InfraID,
+				Namespace: capiutils.Namespace,
+			}, c); err != nil {
+				if apierrors.IsNotFound(err) {
+					return false, nil
+				}
+				return false, err
+			}
+			cluster = c
+			return cluster.Status.InfrastructureReady, nil
+		}); err != nil {
+			return fileList, err
+		}
+		if cluster == nil {
+			return fileList, fmt.Errorf("error occurred during load balancer ready check")
+		}
+		if cluster.Spec.ControlPlaneEndpoint.Host == "" {
+			return fileList, fmt.Errorf("control plane endpoint is not set")
+		}
+	}
+
+	if p, ok := i.capiProvider.(InfraReadyProvider); ok {
+		infraReadyInput := InfraReadyInput{
+			Client:        cl,
+			InstallConfig: installConfig,
+			InfraID:       clusterID.InfraID,
+		}
+
+		if err := p.InfraReady(ctx, infraReadyInput); err != nil {
+			return fileList, fmt.Errorf("failed provisioning resources after infrastructure ready: %w", err)
+		}
+	} else {
+		logrus.Debugf("No infrastructure ready requirements for the %s provider", i.capiProvider.Name())
+	}
+
+	bootstrapIgnData, err := injectInstallInfo(bootstrapIgnAsset.Files()[0].Data)
+	if err != nil {
+		return nil, fmt.Errorf("unable to inject installation info: %w", err)
+	}
+
+	// The cloud-platform may need to override the default
+	// bootstrap ignition behavior.
+	if p, ok := i.capiProvider.(IgnitionProvider); ok {
+		ignInput := IgnitionInput{
+			BootstrapIgnData: bootstrapIgnData,
+			InfraID:          clusterID.InfraID,
+			InstallConfig:    installConfig,
+		}
+
+		if bootstrapIgnData, err = p.Ignition(ctx, ignInput); err != nil {
+			return fileList, fmt.Errorf("failed creating bootstrap ignition secret: %w", err)
```

Comment:
This is not "creating bootstrap ignition secret" any more. I'll leave it to your native-speaker skills to phrase a replacement, because I would use too many words!